src/main/java/org/broadinstitute/hellbender/tools/exome/IntervalBackedExonCollection.java

```diff
@@ -33,46 +63,114 @@
      */
     private int lastBinarySearchResult = -1;
 
+
+
     /**
      * Creates a exon data-base give a sorted list of intervals.
      *
+     * <p>
+     *     Intervals will be sorted by their contig id lexicographical order.
+     * </p>
+     *
      * @param intervals the input interval list. Is assumed to be sorted
      * @throws IllegalArgumentException if {@code intervals} is {@code null}.
      */
-    public IntervalBackedExonCollection(final List<GenomeLoc> intervals) {
+    public IntervalBackedExonCollection(final List<SimpleInterval> intervals) {
+        this(intervals, String::compareTo);
+    }
+
+    /**
+     * Creates a exon data-base give a sorted list of intervals.
+     *
+     * <p>
+     *     The contig name comparator provided is used to establish the absolute order between
+     *     intervals on different contigs.
+     * </p>
+     *
+     * @param intervals the input interval list. Is assumed to be sorted
+     * @param contigComparator comparator used to establish the absolute order
+     *                         between intervals in different contigs.
+     * @throws IllegalArgumentException if {@code intervals} is {@code null}.
+     */
+    public IntervalBackedExonCollection(final List<SimpleInterval> intervals,
+                                        final Comparator<String> contigComparator) {
         if (intervals == null) {
             throw new IllegalArgumentException("the input intervals cannot be null");
         }
-        checkInputIntervalCoherence(intervals);
-        this.sortedIntervals = Collections.unmodifiableList(new ArrayList<>(intervals));
-        this.intervalsByName = composeIntervalsByName(intervals);
+        if (contigComparator == null) {
+            throw new IllegalArgumentException("the contig comparator cannot be null");
+        }
+        if (intervals.contains(null)) {
+            throw new IllegalArgumentException("the input cannot contain null");
+        }
+
+        intervalOverlapComparator = buildIntervalOverlapComparator(contigComparator);
+        intervalStandardComparator = buildIntervalStandardComparator(contigComparator);
+        sortedIntervals = intervals.stream().sorted(intervalStandardComparator).collect(Collectors.toList());
+        checkForOverlaps(sortedIntervals,intervalOverlapComparator);
+        this.intervalsByName = composeIntervalsByName(sortedIntervals);
     }
 
     /**
-     * Throws a exception if there some interval location incoherence.
+     * Fails with an exception if the intervals collection has overlapping intervals.
+     * @param sortedIntervals
+     * @param overlapComparator
+     */
+    private static void checkForOverlaps(final List<SimpleInterval> sortedIntervals,
```

Comment:
Why does this use the overlap comparator instead of just calling `SimpleInterval.overlaps`?